numpy/core/src/multiarray/shape.c

```diff
@@ -111,11 +112,11 @@ PyArray_Resize(PyArrayObject *self, PyArray_Dims *newshape, int refcheck,
             return NULL;
         }
 
-        if (newsize == 0) {
-            sd = PyArray_DESCR(self)->elsize;
+        if (newnbytes == 0) {
```

Comment:
Could do
```
sd = newnbytes ? newnbytes : elsize
```

Reply:
Would prefer to keep the `== 0`, because that's handling the special meaning of `0` which is "allow resizing".

Could flatten the `if` though